Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -0,0 +1,119 @@
+            let paymentTitle = "Payment Pending"
+                paymentMessage = "Payment of Rs. " <> show driverFee.totalAmount.getMoney <> " is pending. Pay by " <> show driverFee.payBy <> " to avoid being blocked."
+            Notify.sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.PAYMENT_PENDING paymentTitle paymentMessage driver.id driver.deviceToken
+      Esq.runTransaction $ do
+        mapM_ (updateStatus PAYMENT_PENDING) driverFeeIds
+      ReSchedule <$> getRescheduledTime
+    else do
+      logInfo "No ongoing payment found."
+      return Complete
+
+unsubscribeDriverForPaymentOverdue ::
+  ( MonadFlow m,
+    HedisFlow m r,
+    CoreMetrics m,
+    HasCacheConfig r,
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r
+  ) =>
+  Job 'UnsubscribeDriverForPaymentOverdue ->
+  m ExecutionResult
+unsubscribeDriverForPaymentOverdue Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  let jobData = jobInfo.jobData
+      startTime = jobData.startTime
+  now <- getCurrentTime
+  (driverFees, driverFeeIds, feeZipDriver, relevantDriverIds) <- calcDriverFeeAttr PAYMENT_PENDING startTime now
+  if not $ null driverFees
+    then do
+      for_ feeZipDriver $ \(driverFee, mbDriver) -> do
+        case mbDriver of
+          Nothing -> do
+            logInfo "Driver Not found. This should not be possible."
+            throwError (InternalError "Driver Not Found") -- Unreachable
+          Just driver -> do
+            let paymentTitle = "Blocked - Payment Overdue"
+                paymentMessage = "Sorry, you are blocked from taking rides as payment of Rs. " <> show driverFee.totalAmount.getMoney <> " is outstanding. Please pay the balance to get unblocked."
+            Notify.sendNotificationToDriver driver.merchantId FCM.SHOW Nothing FCM.PAYMENT_OVERDUE paymentTitle paymentMessage driver.id driver.deviceToken
+      Esq.runTransaction $ do
+        mapM_ (updateStatus PAYMENT_OVERDUE) driverFeeIds
+        mapM_ (updateSubscription False) relevantDriverIds
+    else do
+      logInfo "No ongoing payment found."
+  return Complete
+
+calcDriverFeeAttr :: (MonadFlow m, EsqDBFlow m r, Esq.EsqDBReplicaFlow m r) => DriverFeeStatus -> UTCTime -> UTCTime -> m ([DriverFee], [Id DriverFee], [(DriverFee, Maybe Person)], [Id Driver])
+calcDriverFeeAttr driverFeeStatus startTime endTime = do
+  driverFees <- findFeesInRangeWithStatus startTime endTime driverFeeStatus
+  let driverFeeIds = (.id) <$> driverFees
+      relevantDriverIds = (.driverId) <$> driverFees
+  relevantDrivers <- mapM (Esq.runInReplica . QPerson.findById) (cast <$> relevantDriverIds)
+  let feeZipDriver = zip driverFees relevantDrivers
+  return (driverFees, driverFeeIds, feeZipDriver, relevantDriverIds)
+
+getRescheduledTime :: (MonadFlow m) => m UTCTime
+getRescheduledTime = do
+  addUTCTime driverPaymentReminderInterval <$> getCurrentTime
+
+driverPaymentReminderInterval :: NominalDiffTime
+driverPaymentReminderInterval = 1800
```

Comment:
can this also be a config ?

Reply:
I'll add all these in TC instead of making it config

Reply:
transporterConfig right ? thats what i meant